front/pages/w/[wId]/u/gens/index.tsx

```diff
@@ -807,25 +807,36 @@ export default function AppGens({
     // console.log(textWithCursor);
 
     // turn genDocumentExtracts into an array of extracts ordered by score
-    const extracts = retrieved
+    const potentialExtracts = retrieved
       .map((d) => {
         const chunks = d.chunks.sort((a, b) => a.offset - b.offset);
         const text = chunks.map((c) => c.text).join("");
         return {
           documentId: d.documentId,
           text: text,
           score: d.llm_score || 0,
+          tokenCount: d.tokenCount,
         };
       })
-      .sort((a, b) => a.score - b.score)
-      .map((e) => {
-        return {
-          documentId: e.documentId,
-          extract: e.text,
-          score: e.score.toFixed(2),
-        };
-      });
+      .sort((a, b) => a.score - b.score);
 
+    const extracts: {
+      documentId: string;
+      extract: string;
+      score: string;
+    }[] = [];
+
+    potentialExtracts.reduce((space, d) => {
+      if (d.tokenCount <= space) {
+        extracts.push({
+          documentId: d.documentId,
+          extract: d.text,
+          score: d.score.toFixed(2),
+        });
+        space -= d.tokenCount;
+      }
+      return space;
+    }, 8000);
```

Comment:
Maybe a bit less? if we account for prompt + noise around chunks we may overflow here no?

Reply:
7000?

Reply:
Sure let's do multiple of 1024 so 7168